Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptIDLookup.java

```diff
@@ -95,4 +102,22 @@
         
         return new ResponseEntity<String>(msg.getXML(xmlEntries), HttpStatus.OK);
     }
+
+    private void addAlternativeIds(String id, ConceptEntry entry) throws LuceneException {
+        if (id.contains("??")) {
+            // User have queried with a general wordnet id
+            // Added general wordnet id, CP id and specific wordnet id
+            entry.getAlternativeIds().add(id);
+            entry.getAlternativeIds().add(entry.getId());
+        } else if (!id.startsWith("CON")) {
```

Comment:
I would not use this condition for the check. This is something we could change in the future (even if it's not extremely likely, but it's not unimaginable). The wordnet ids, however, for sure won't change. Also, I would put logic to decide what kind of id it is into a service class so that it can be reused by user classes.